src/mpid/ch4/netmod/ofi/ofi_coll_select_container_parsers.c

```diff
@@ -0,0 +1,114 @@
+
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2016 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+
+#include <mpidimpl.h>
+#include "ofi_noinline.h"
+
+const char *MPIDU_SELECTION_Bcast_ofi[] = {
+    "binomial",
+    "scatter_recursive_doubling_allgather",
+    "scatter_ring_allgatherv",
+};
```

Comment:
Again, nothing to do with `ofi`... :)

Reply:
These are MPIR algorithms that work from any layer. A good example would be the POSIX list. It has `release-gather` device specific algorithm and other MPIR algorithms which can also be run from device layer